reach/reach.py

```diff
@@ -628,6 +628,48 @@ def _most_similar_batch(
                 word_index = np.flip(np.argsort(sims_for_word))
                 yield [(self.indices[indices[idx]], sims_for_word[idx]) for idx in word_index]
 
+    def indices_threshold(
+        self,
+        vectors: npt.NDArray,
+        threshold: float = 0.5,
+        batch_size: int = 100,
+        show_progressbar: bool = False,
+    ) -> Iterator[np.ndarray]:
+        """
+        Retrieve the unsorted indices of the nearest neighbors for the given vectors above a threshold.
+
+        This method is useful if you want to find the indices of the most similar items above a certain threshold
+        but don't need them to be sorted, which is a time consuming operation.
+
+        :param vectors: The vectors to find the most similar vectors to.
+        :param threshold: The threshold to use.
+        :param batch_size: The batch size to use. 100 is a good default option. Increasing the batch size may increase
+            the speed.
+        :param show_progressbar: Whether to show a progressbar.
+
+        :return: For each item in the input, the indices of the nearest neighbors are returned as an array.
+        """
+        vectors = np.array(vectors)
+        if np.ndim(vectors) == 1:
+            vectors = vectors[None, :]
+
+        return self._indices_threshold_batch(vectors, batch_size, threshold, show_progressbar)
+
+    def _indices_threshold_batch(
```

Comment:
this helper function only exists because we have `most_similar` and `nearest_neighbor` thresholding function. If you don't plan on adding the item counterpart, I would just lose this one.